### ISSUE TITLE
Reindex MTZ using python

### ISSUE THREAD

**Issue by FilomenoSanchez** [project-gemmi/gemmi#87]
Hi @wojdyr, thanks for the package, I find it really useful. I was wondering if there is a way to change the spacegroup of a `gemmi.Mtz` instance in python (including the reindexing of the reflections). For example, if I had a MTZ in `P321` and I want to change it to `P 31 2 1` I know it would possible to do `mtz.spacegroup = gemmi.SpaceGroup('P 31 2 1')` but I have the impression this is just changing some attribute in `gemmi.Mtz` and not reindexing the reflections. Thanks!

**Comment by wojdyr** [comment]
My guess would be that in this case reindexing is not necessary, but I don't know these things well.
When I needed to reindex MTZ files in the past I used CCP4 program [aimless](http://legacy.ccp4.ac.uk/html/aimless.html).

**Comment by FilomenoSanchez** [comment]
Hi, I'm not sure as well if it would be necessary to reindex reflections in that particular case (`P321` -> `P 31 2 1`) it was just an example really. I've also noticed it is possible to do this using CCP4 [reindex](http://www.sb.fsu.edu/~xray/Manuals/CCP4/reindex.html), but I was just wondering if there was a way to do this through python so that I can avoid having to make  a system call with `subprocess.Popen()`. The reason why I thought this would be possible is because I had the impression that `gemmi reindex` would do what I want (not 100% sure about this though, I didn't have the time to look into it properly) and so I thought that there might be a way to do this through python using gemi.

**Comment by wojdyr** [comment]
CCP4 reindex requires specifying the reindexing operator. If it just applies this operator to all Miller indices this would be trivial to implement.

**Comment by FilomenoSanchez** [comment]
It would definitely be very useful to have something like this implemented in gemmi.

**Comment by wojdyr** [comment]
This week I was looking into reindexing again.

I didn't remember that I started writing `gemmi reindex` a year ago. @FilomenoSanchez wrote about it, but I either overlooked it or forgot about it since March. Anyway, I found `gemmi reindex`, but it wasn't working correctly. I fixed a few bugs and perhaps in some cases it works correctly now, but for sure not in all cases. It requires much more work and testing.

There are many things that reindexing program needs to change in MTZ file:
- Miller indices of reflections
- space group
- unit cell parameters
- phases (by adding phase shift)
- phase probability coefficients (Hendrickson-Lattman) – I didn't look into it yet
- sometimes I(+)/I(-) and other (+)/(-) columns need to be swapped (when the reindexing operator changes hand?)
- reflections need to be removed if reindexing gives non-integral Miller indices
- and I suppose there are things I overlooked

I'd be also good to determine if reindexing with given operator makes sense.

I think the biggest part of the work would be setting up tests that would check all these things with reindexing operators that cover all corner cases.

Moving reindexing code to the library and adding Python bindings would be the simplest part and I can do it at any point if anyone needs it, but currently it'd be usable only for testing.
